tests/histogram.rs

```diff
@@ -395,3 +398,50 @@ fn scaled_set_to() {
     h2.set_to(&h1).unwrap();
     are_equal(&h1, &h2);
 }
+
+
+#[test]
+fn random_write_full_value_range_precision_5_no_panic() {
+    let mut h = Histogram::<u64>::new_with_bounds(1, u64::max_value(), 5).unwrap();
+
+    let mut rng = rand::weak_rng();
+
+    for _ in 0..1_000_000 {
+        let mut r: u64 = rng.gen();
+        if r == 0 {
+            r = 1;
+        }
+
+        h.record(r).unwrap();
+    }
+}
+
+
+#[test]
+fn random_write_full_value_range_precision_0_no_panic() {
+    let mut h = Histogram::<u64>::new_with_bounds(1, u64::max_value(), 0).unwrap();
+
+    let mut rng = rand::weak_rng();
+
+    for _ in 0..1_000_000 {
+        let mut r: u64 = rng.gen();
+        if r == 0 {
+            r = 1;
+        }
+
+        h.record(r).unwrap();
+    }
+}
+
+#[test]
+fn random_write_middle_of_value_range_precision_3_no_panic() {
+    let low = 1_000;
+    let high = 1_000_000_000;
+    let mut h = Histogram::<u64>::new_with_bounds(low, high, 3).unwrap();
+
+    let mut rng = rand::weak_rng();
+
+    for _ in 0..1_000_000 {
```

Comment:
Oh, you're right, I misread the code. It seems fine the way it stands. The `low` and `high` arguments to `gen_range` below is what really matters.